tests/unit/s2n_examples_test.c

```diff
@@ -233,19 +233,25 @@ static S2N_RESULT s2n_run_self_talk_test(s2n_test_scenario scenario_fn)
 
     pid_t client_pid = fork();
     if (client_pid == 0) {
-        /* Suppress stdout when running the examples.
-         * This only affects the new client process.
+        /* use a new scope to force DEFER_CLEANUP functions to run before the
+         * exit system call
          */
```

Comment:
Currently uncertain. _something_ changed in the s2n_x509_validator, probably related to memory freeing? If it is the case that something in `validator_wipe` was causing this issue, then it makes sense that only the client experienced the issue because it's the only agent that needs to call validator wipe. I created #4363 to track further investigation.

Reply:
Added an explicit free for now, and working on a full fix in #4369